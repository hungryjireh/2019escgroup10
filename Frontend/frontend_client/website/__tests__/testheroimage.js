import React from "react";
import { render, cleanup, fireEvent, waitForElement, getByLabelText, getByText, getByTestId } from 'react-testing-library';
import HeroImage from "../components/Hero/HeroImage";
import {HeroImageMobile, HeroImageWrapper, TextHighlight, TelegramSubtext, ImageSubtext, ImageHeader} from "../components/Hero/HeroImage";

afterEach(cleanup);

it("HeroImageMobile renders", () => {
    const { asFragment } = render(<HeroImageMobile />);
    expect(asFragment()).toMatchSnapshot();
});

it("HeroImageWrapper renders", () => {
    const { asFragment } = render(<HeroImageWrapper />);
    expect(asFragment()).toMatchSnapshot();
});

it("TextHighlight renders", () => {
    const { asFragment } = render(<TextHighlight />);
    expect(asFragment()).toMatchSnapshot();
});

it("TelegramSubtext renders", () => {
    const { asFragment } = render(<TelegramSubtext />);
    expect(asFragment()).toMatchSnapshot();
});

it("ImageSubtext renders", () => {
    const { asFragment } = render(<ImageSubtext />);
    expect(asFragment()).toMatchSnapshot();
});

it("ImageHeader renders", () => {
    const { asFragment } = render(<ImageHeader />);
    expect(asFragment()).toMatchSnapshot();
});

test("Test HeroImage Components", () => {
    const { getByAltText, getByText } = render(<HeroImage />);
    expect(getByAltText("bricks"));
    expect(getByAltText("arm"));
    expect(getByAltText("hero image robot"));
    expect(getByText("Chat"));
    expect(getByText("With Us Now."));
    expect(getByText(/Say/));
    expect(getByText("Hello"));
    expect(getByText(/to Botty!/));
    expect(getByText("Also available on:"));
    expect(getByAltText("telegram"));
});

test("Telegram Link renders", () => {
    const { getByText } = render(<HeroImage />);
    expect(getByText("Also available on:").innerHTML.includes("https://telegram.me/acnapibotty_bot")).toBeTruthy;
});