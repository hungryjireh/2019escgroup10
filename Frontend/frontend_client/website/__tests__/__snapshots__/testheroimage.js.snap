// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImageMobile renders 1`] = `
<DocumentFragment>
  <img
    class="sc-htpNat dSGxGw"
  />
</DocumentFragment>
`;

exports[`HeroImageWrapper renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa hqwBfd"
  />
</DocumentFragment>
`;

exports[`ImageHeader renders 1`] = `
<DocumentFragment>
  <h1
    class="sc-bxivhb cCxSJf"
  />
</DocumentFragment>
`;

exports[`ImageSubtext renders 1`] = `
<DocumentFragment>
  <h2
    class="sc-ifAKCX iPLhhO"
  />
</DocumentFragment>
`;

exports[`TelegramSubtext renders 1`] = `
<DocumentFragment>
  <div
    class="sc-EHOje gUgiRc"
  />
</DocumentFragment>
`;

exports[`TextHighlight renders 1`] = `
<DocumentFragment>
  <span
    class="sc-bZQynM hPNeBX"
  />
</DocumentFragment>
`;
